### PR TITLE
Add deduplication to backend

### LINKED ISSUE #44: De-duplicate based on unique smiles
Results in the frontend should be deduplicated so that each unique smiles has 1 row with multiple IDs from source databases.

### PR DESCRIPTION
**Pull request by lorton**
Also added a single threaded argument for easily debugging creation of database issues, and deleted an outdated comment at top of tests.

Fixes #44

### CODE REVIEW

File: gpusim.cpp [hunk 1]
```diff
@@ -315,16 +316,27 @@ void GPUSimServer::incomingSearchRequest()
...
     QDataStream ids_stream(&output_ids, QIODevice::WriteOnly);
     QDataStream scores_stream(&output_scores, QIODevice::WriteOnly);
+    int dedup_count = 0;
     for (unsigned int i = 0; i < results_smiles.size(); i++) {
+        std::stringstream ss;
+        ss << results_ids[i];
         smiles_stream << results_smiles[i];
-        ids_stream << results_ids[i];
+        while(i+1 < results_smiles.size()) {
+            if(strcmp(results_smiles[i], results_smiles[i+1])!=0) break;
```
Comment (cdvonbargen) [inline gpusim.cpp]: eh strcmp.

if (std::string(results_smiles[i]) != std::string(results_smiles[i+1])) {
    break;
}

File: gpusim.cpp [hunk 2]
```diff
@@ -315,16 +316,27 @@ void GPUSimServer::incomingSearchRequest()
...
+            ss << ";:;";
+            ss << results_ids[i+1];
+            i++;
```
Comment (cdvonbargen) [inline gpusim.cpp]: ss << ";:;" << results_id[++i];

File: python/gpusim_createdb.py [hunk 3]
```diff
@@ -95,7 +97,10 @@ def writeData(self, qds):
 
 
 def main():
+    global dview
     args = parse_args()
+    if args.singleThreaded:
+        dview = None
     qf = QtCore.QFile(args.outputfile)
```
Comment (cdvonbargen) [inline python/gpusim_createdb.py]: blah. I guess I would have preferred having the ipyparallel try import block assign ipp = None in the except, and then move the dview extraction to main

Comment (lorton) [reply python/gpusim_createdb.py]: Good suggestion

File: gpusim.cpp (hunk 2 above)
Comment (lorton) [reply gpusim.cpp]: Done

**Pull request merged**